ckanext/cnra_schema/helpers.py

```diff
@@ -69,3 +73,48 @@ def set_waf_contact_point(package_dict, iso_values, contact_point_mapping):
         delete_existing_extra_from_package_dict(email_field, package_dict)
 
     return package_dict
+
+
+def is_cnra_schema_field_populated(package_dict, field):
+    field_name = field.get('field_name')
+
+    if package_dict.get(field_name) and field_name not in 'spatial_details' \
+            and field.get('preset') in ['composite', 'composite_repeating']:
+
+        subfield_literal_eval = literal_eval(package_dict.get(field_name))
+
+        return is_dict_populated(subfield_literal_eval)
+
+    """"
+    NOTE: This function returns True due to the front-end calling this helper function as an AND in the conditional.
+    If this function returns False, the whole condition would be false and non-composite fields would not be included
+    in the "Additional Metadata" table.
+    """
+    return True
+
+
+def is_dict_populated(package_dict_field):
+    is_subfield_empty = False
+
+    if package_dict_field and isinstance(package_dict_field, dict):
+        return is_sub_dict_populated_recursively_impl(package_dict_field, is_subfield_empty)
+    elif isinstance(package_dict_field, list):
+        is_subfield_list_populate = [is_sub_dict_populated_recursively_impl(x, is_subfield_empty)
+                                     for x in package_dict_field]
+        return True in is_subfield_list_populate
+    else:
+        return False
```

Comment:
The final else statement isn't needed. `is_dict_populated` should just return False by default.